qml/Components/Dialogs.qml

```diff
@@ -34,12 +34,21 @@ MouseArea {
     // to be set from outside, useful mostly for testing purposes
     property var unitySessionService: DBusUnitySessionService
     property var closeAllApps: function() {
+        // FIXME! This is an awefull way of doing this!
+        var i = 0;
```

Comment:
Shouldn't `i` count down from the length to zero?

Reply:
there is no way of getting the list or length of the amount of apps in the current api. This just counts up until it hits NULL then resets it, and does this 20 times, if not it just continues reboot/shutdown

Reply:
Wouldn't this happen near-instantly on most devices? From what I've seen, Unity8 can ask apps to stop *really* fast.